src/Assert.php

```diff
@@ -27,188 +27,7 @@
- * @method static void allNotWhitespaceOnly($values, $message = '')
- * @method static void allStartsWith($values, $prefix, $message = '')
- * @method static void allNotStartsWith($values, $prefix, $message = '')
- * @method static void allStartsWithLetter($values, $message = '')
- * @method static void allEndsWith($values, $suffix, $message = '')
- * @method static void allNotEndsWith($values, $suffix, $message = '')
- * @method static void allRegex($values, $pattern, $message = '')
- * @method static void allNotRegex($values, $pattern, $message = '')
- * @method static void allUnicodeLetters($values, $message = '')
- * @method static void allAlpha($values, $message = '')
- * @method static void allDigits($values, $message = '')
- * @method static void allAlnum($values, $message = '')
- * @method static void allLower($values, $message = '')
- * @method static void allUpper($values, $message = '')
- * @method static void allLength($values, $length, $message = '')
- * @method static void allMinLength($values, $min, $message = '')
- * @method static void allMaxLength($values, $max, $message = '')
- * @method static void allLengthBetween($values, $min, $max, $message = '')
- * @method static void allFileExists($values, $message = '')
- * @method static void allFile($values, $message = '')
- * @method static void allDirectory($values, $message = '')
- * @method static void allReadable($values, $message = '')
- * @method static void allWritable($values, $message = '')
- * @method static void allClassExists($values, $message = '')
- * @method static void allSubclassOf($values, $class, $message = '')
- * @method static void allInterfaceExists($values, $message = '')
- * @method static void allImplementsInterface($values, $interface, $message = '')
- * @method static void allPropertyExists($values, $property, $message = '')
- * @method static void allPropertyNotExists($values, $property, $message = '')
- * @method static void allMethodExists($values, $method, $message = '')
- * @method static void allMethodNotExists($values, $method, $message = '')
- * @method static void allKeyExists($values, $key, $message = '')
- * @method static void allKeyNotExists($values, $key, $message = '')
- * @method static void allValidArrayKey($values, $message = '')
- * @method static void allCount($values, $key, $message = '')
- * @method static void allMinCount($values, $min, $message = '')
- * @method static void allMaxCount($values, $max, $message = '')
- * @method static void allCountBetween($values, $min, $max, $message = '')
- * @method static void allIsList($values, $message = '')
- * @method static void allIsNonEmptyList($values, $message = '')
- * @method static void allIsMap($values, $message = '')
- * @method static void allIsNonEmptyMap($values, $message = '')
- * @method static void allUuid($values, $message = '')
- * @method static void allThrows($expressions, $class = 'Exception', $message = '')
+ * @mixin Mixin
```

Comment:
I don't think the ecosystem is ready for this - PhpStorm and PHPStan don't understand this yet.

Reply:
PHPStorm correctly jumps to definition and infers correct types here, from what I see in my environment.

This stuff has been used in the (terrible, but at least auto-completing) illuminate library for ages: https://github.com/illuminate/database/commit/9e155ee7cc2bbd90b1a749458f989b0aa8a8cc1e

Reply:
You might be right, I knew it’s an obscure annotation and it didn’t work a few years back. Time to add support to PHPStan I guess 😊

Reply:
@ondrejmirtes do you prefer us to duplicate the docblocks? that could work, in theory (need to still write those tests 😅)

Reply:
I have an interesting use-case for `@mixin` in regard to generics - it will allow describing complex scenarios like `RecursiveIteratorIterator` (https://github.com/phpstan/phpstan/issues/466), or Prophecy (https://github.com/phpspec/prophecy). So I want to tackle this, this will make me do it sooner :)

Reply:
After we get an approval from @BackEndTea I will need couple more evenings to finish tests, other than that - this feature is ready.

Reply:
PHPStan's dev-master now supports `@mixin`: https://github.com/phpstan/phpstan-src/commit/b02ee1444440ed995c17171c456feec2c56e0cb2

Will probably release it after this weekend 👍

Reply:
awesome work!💃